### ISSUE TITLE
Resource group lock prevents node instances from being redeployed/reimaged

### ISSUE THREAD

**Issue by templier2** [Ontotext-AD/terraform-azure-graphdb#98]
**Describe the bug**
After reimaging an instance it can't start with this error in log:

```
2024-11-27 13:58:14: Updating IP address for node-2
ERROR: (ScopeLocked) The scope '/subscriptions/<subscrID>/resourceGroups/rg-graphdb-preprod/providers/Microsoft.Network/pri
vateDnsZones/graphdb-preprod.dns.zone/A/node-2' cannot perform delete operation because following scope(s) are locked: '/subscriptions/<subscrID>/resourceGroups/rg-graphdb-preprod'. Please remove the lock and try again.
Code: ScopeLocked
Message: The scope '/subscriptions/<subscrID>/resourceGroups/rg-graphdb-preprod/providers/Microsoft.Network/privateDnsZones
/graphdb-preprod.dns.zone/A/node-2' cannot perform delete operation because following scope(s) are locked: '/subscriptions/<subscrID>/resourceGroups/rg-graphdb-preprod'. Please remove the lock and try again.
ERROR: (BadRequest) The list of record sets of type 'A' may not contain multiple entries with the same 'ipv4Address'.
Code: BadRequest
Message: The list of record sets of type 'A' may not contain multiple entries with the same 'ipv4Address'.
```

Possibly, it could happen after setting TF variable:
graphdb_java_options = "-Dgraphdb.page.cache.size=5G"

**To Reproduce**
After deployment try to reimage any instance from VMSS.

```hcl
// paste code snippets here
```

**Expected behavior**
An instance is redeploying successfully

**Nice to have**
- [ ] Terminal output
- [ ] Screenshots
- [ ] OS and Terraform versions

**Additional context**

**Comment by simonzhekoff** [comment]
Hello, @templier2!

Thanks for raising the bug. 

I want to inform you that we are working on a fix, regarding this issue. 

While you are waiting, you can change the lock_resources variable to false, in order to temporary fix the issue. 

Best Regards,
Simeon Zhekov
Ontotext AD
DevOps Engineer

**Comment by simonzhekoff** [comment]
Hello, @templier2 bug has been fixed and it will be released with v1.5.0 of the module.

We will close the issue now.

Best Regards,
Simeon Zhekov
DevOps Engineer
Ontotext AD
